GDASApp develop build fails with error in utils/soca/gdas_incr_handler.cc

Clone GDASApp `develop` at 9fa348f on Orion inside g-w.  Execution of g-w `build_gdas.sh` aborts with
```
8-xgstw63/include -isystem /work/noaa/epic-ps/role-epic-ps/spack-stack/spack-stack-1.3.1/envs/unified-env/install/intel/2022.0.2/hdf5-1.12.2-eefcl5d/include -O2 -g -DNDEBUG -fPIE -qopenmp -std=c++17 -MD -MT gdas-utils/soca/CMakeFiles/gdas_incr_handler.x.dir/gdas_incr_handler.cc.o -MF CMakeFiles/gdas_incr_handler.x.dir/gdas_incr_handler.cc.o.d -o CMakeFiles/gdas_incr_handler.x.dir/gdas_incr_handler.cc.o -c /work2/noaa/da/rtreadon/git/global-workflow/test2/sorc/gdas.cd/gdas-utils/soca/gdas_incr_handler.cc
In file included from /work2/noaa/da/rtreadon/git/global-workflow/test2/sorc/gdas.cd/gdas-utils/soca/gdas_incr_handler.h(25),
                 from /work2/noaa/da/rtreadon/git/global-workflow/test2/sorc/gdas.cd/gdas-utils/soca/gdas_incr_handler.cc(1):
/work2/noaa/da/rtreadon/git/global-workflow/test2/sorc/gdas.cd/gdas-utils/soca/gdas_postprocincr.h(165): error: no instance of constructor "soca::LinearVariableChange::LinearVariableChange" matches the argument list
            argument types are: (const soca::Geometry, soca::LinearVariableChangeParameters)
      soca::LinearVariableChange lvc(this->geom_, params);
                                     ^
/work2/noaa/da/rtreadon/git/global-workflow/test2/sorc/gdas.cd/soca/src/soca/../soca/LinearVariableChange/LinearVariableChange.h(44): \
note: this candidate was rejected because mismatch in count of arguments
  class LinearVariableChange : public util::Printable {
        ^
/work2/noaa/da/rtreadon/git/global-workflow/test2/sorc/gdas.cd/soca/src/soca/../soca/LinearVariableChange/LinearVariableChange.h(55): \
note: this candidate was rejected because arguments do not match
    explicit LinearVariableChange(const Geometry &, const eckit::Configuration &);
             ^

compilation aborted for /work2/noaa/da/rtreadon/git/global-workflow/test2/sorc/gdas.cd/gdas-utils/soca/gdas_incr_handler.cc (code 2)
make[2]: *** [gdas-utils/soca/CMakeFiles/gdas_incr_handler.x.dir/gdas_incr_handler.cc.o] Error 2
```
The `gdas-utils` directory is a soft link to `utils`.  Thus, [`gdas.cd/gdas-utils/soca/gdas_incr_handler.cc`](https://github.com/NOAA-EMC/GDASApp/blob/develop/utils/soca/gdas_incr_handler.cc) is a GDASApp code.  

As a test comment out the following lines in [`utils/soca/CMakeLists.txt`](https://github.com/NOAA-EMC/GDASApp/blob/develop/utils/soca/CMakeLists.txt)

 ```
# Increment post processing
-ecbuild_add_executable( TARGET gdas_incr_handler.x
-                        SOURCES gdas_incr_handler.cc gdas_postprocincr.h)
-target_compile_features( gdas_incr_handler.x PUBLIC cxx_std_17)
-target_link_libraries( gdas_incr_handler.x PUBLIC NetCDF::NetCDF_CXX oops atlas soca)
+#ecbuild_add_executable( TARGET gdas_incr_handler.x
+#                        SOURCES gdas_incr_handler.cc gdas_postprocincr.h)
+#target_compile_features( gdas_incr_handler.x PUBLIC cxx_std_17)
+#target_link_libraries( gdas_incr_handler.x PUBLIC NetCDF::NetCDF_CXX oops atlas soca)
```

With the above change g-w `build_gdas.sh` runs to completion.

It looks like something changed in soca which requires a corresponding change in `/gdas_incr_handler.cc`.

--- a/utils/soca/gdas_postprocincr.h
+++ b/utils/soca/gdas_postprocincr.h
@@ -159,10 +159,9 @@ class PostProcIncr {
       return socaIncr;
     }
     oops::Log::info() << "======      applying specified change of variables" << std::endl;
-    soca::LinearVariableChangeParameters params;
-    params.deserialize(lvcConfig_);
-    oops::Log::info() <<  params << std::endl;
-    soca::LinearVariableChange lvc(this->geom_, params);
+    oops::Log::info() <<  lvcConfig_ << std::endl;
+    eckit::LocalConfiguration lvcConfig_;
+    soca::LinearVariableChange lvc(this->geom_, lvcConfig_);
     oops::Log::info() <<  "traj: " << xTraj_ << std::endl;
     lvc.changeVarTraj(xTraj_, socaIncrVar_);
     lvc.changeVarTL(socaIncr, socaIncrVar_);

